ghost/core/core/server/services/members/jobs/index.js

```diff
@@ -19,9 +22,31 @@ module.exports = {
                 name: 'clean-expired-comped'
             });
 
-            hasScheduled = true;
+            hasScheduled.expiredComped = true;
+        }
+
+        return hasScheduled.expiredComped;
+    },
+
+    async scheduleTokenCleanupJob() {
+        if (
+            !hasScheduled.tokens &&
+            !process.env.NODE_ENV.startsWith('test')
+        ) {
+            // use a random seconds/minutes/hours value to avoid delete spikes to the database
+            const s = Math.floor(Math.random() * 60); // 0-59
```

Comment:
I've seen this randomized time for cron logic pop up in multiple places. Just a note to self it might be something we should extract into a small utility 🙃 

What was a thought process here to randomize the "hour" parameter? Do we need this job to be spread out through the day?

The reference point I've found was, for example, the complimentary plan cleanup runs at `12:05:X AM`. 

No personal opinion here, I'm trying to see what is the usecase and common patterns for job scheduling we are developing.

Reply:
It doesn't really matter at what time this job runs, it just needs to run once a day. So if we are going to spread the execution of the jobs, then it is better to spread them across the whole day I think. I don't think there is an advantage to know that we run it at approximately a given hour.